app/domain/logs.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'util/log_message_with_code_class'
+
+unless defined? LogMessages::Authentication::Security::SecurityValidated
+  # this wrapper is here so these classes will not be loaded by Rails
+  # auto-load. #TODO: fix this in a proper manner
+
+  module LogMessages
+
+    module Authentication
+
+      module Security
+
+        SecurityValidated = ::Util::LogMessageWithCodeClass.new(
+          msg: "Security validated",
+          code: "CONJ00001D"
+        )
+
+        UserNotAuthorized = ::Util::LogMessageWithCodeClass.new(
+          msg: "User '{0}' is not authorized to " \
+          "authenticate with webservice '{1}'",
+          code: "CONJ00002D"
+        )
+
+      end
+
+      module Origin
+
+        OriginValidated = ::Util::LogMessageWithCodeClass.new(
+          msg: "Origin validated",
+          code: "CONJ00003D"
+        )
+
+      end
+
+      module AuthnOidc
+
+        ExtractedUsernameFromIDToked = ::Util::LogMessageWithCodeClass.new(
+          msg: "Extracted username '{0}' from ID Token",
```

Comment:
Can you please log also "according to id-token-user-property {1}" and give it the value?